src/native-automation/client/input.ts

```diff
@@ -4,20 +4,35 @@ import { SimulatedKeyInfo } from './key-press/utils';
 import { DispatchEventFn } from './types';
 import CDPEventDescriptor from './event-descriptor';
 import { Modifiers } from '../../test-run/commands/options';
+import hammerhead from './deps/hammerhead';
 
 export default class NativeAutomationInput {
     private readonly _dispatchEventFn: DispatchEventFn;
+    private readonly _touchMode: boolean;
     constructor (dispatchEventFn: DispatchEventFn) {
         this._dispatchEventFn = dispatchEventFn;
+        this._touchMode       = hammerhead.utils.featureDetection.isTouchDevice;
+    }
+
+    private async _executeTouchEvent (touchEvent: string, options: any): Promise<void> {
+        const eventOptions = await CDPEventDescriptor.createTouchEventOptions(touchEvent, options);
+
+        return this._dispatchEventFn.single(EventType.Touch, eventOptions);
     }
 
     public async mouseDown (options: any): Promise<void> {
+        if (this._touchMode)
+            return this._executeTouchEvent('touchStart', options);
+
         const eventOptions = await CDPEventDescriptor.createMouseEventOptions('mousePressed', options);
 
         return this._dispatchEventFn.single(EventType.Mouse, eventOptions);
     }
```

Comment:
please check if we can exctract all mouse related code to the separate `_executeMouseEvent` method